src/mdns/mdns_avahi.cpp

```diff
@@ -1167,17 +1167,16 @@ void PublisherAvahi::HostSubscription::HandleResolveResult(AvahiRecordBrowser *
     OTBR_UNUSED_VARIABLE(aFlags);
 
     Ip6Address address;
-    bool       resolved = false;
+    bool       resolved   = false;
+    int        avahiError = 0;
 
     VerifyOrExit(aEvent == AVAHI_BROWSER_NEW);
+    VerifyOrExit(aSize == 16, otbrLogErr("unexpected address data length: %u", aSize),
+                 avahiError = AVAHI_ERR_INVALID_ADDRESS);
```

Comment:
```suggestion
    VerifyOrExit(aSize == 16 || aSize == 4, otbrLogErr("unexpected address data length: %u", aSize),
                 avahiError = AVAHI_ERR_INVALID_ADDRESS);
    VerifyOrExit(aSize == 16, otbrLogErr("IPv4 address ignored"),
                 avahiError = AVAHI_ERR_INVALID_ADDRESS);
```
nit: may have better log for IPv4 address